test-app/app/services/json-storage.ts

```diff
@@ -0,0 +1,48 @@
+import { type ServiceFactory, provide, service } from 'ember-polaris-service';
+import JSONStorage, { type Deserializer } from 'ember-storage-json';
+import Storage from './storage';
+
+export class AppJSONStorage extends JSONStorage {
+  storage = service(this, Storage);
```

Comment:
Injects the store as is required – also pulls in the module dependency naturally